src/functionality.js

```diff
@@ -0,0 +1,124 @@
+        icon.classList.remove('fa-solid');
+        icon.classList.remove('fa-ellipsis-vertical');
+        icon.classList.add('fa-regular');
+        icon.classList.add('fa-trash-can');
+        icon.style.cursor = 'pointer';
+
+        icon.addEventListener('click', () => {
+          const index = parseInt(icon.parentElement.getAttribute('data-index'), 10);
+          this.removeTask(index);
+          icon.parentElement.remove();
+        });
+      });
+
+      item.addEventListener('mouseout', () => {
+        item.style.backgroundColor = 'white';
+      });
+
+      taskInput.addEventListener('focusout', () => {
+        item.style.backgroundColor = 'white';
+        icon.classList.add('fa-solid');
+        icon.classList.add('fa-ellipsis-vertical');
+        icon.classList.remove('fa-regular');
+        icon.classList.remove('fa-trash-can');
+        icon.style.cursor = 'move';
+      });
+
+      taskInput.addEventListener('keypress', (e) => {
+        if (!e) e = window.event;
+        const keyCode = e.code || e.key;
+        if (keyCode === 'Enter') {
+          e.preventDefault();
+          const index = taskInput.parentElement.getAttribute('data-index');
+          this.tasksList[index].description = taskInput.value;
+          localStorage.setItem('tasks', JSON.stringify(this.tasksList));
+        }
+      });
+
+      taskInput.addEventListener('keyup', (e) => {
+        if (e.keyCode === 13) {
+          e.preventDefault();
+          e.target.blur();
+        }
+      });
+    } else {
+      item.innerHTML = `<button><i class="fa-regular fa-square-check checkbox checkbox-ticked"></i></button>
+      <span class="task-span">${taskObj.description}</span>
+      <i class="fa-solid fa-ellipsis-vertical trash-can"></i>`;
+    }
+
+    const checkbox = item.querySelector('.checkbox');
+    const index = parseInt(item.getAttribute('data-index'), 10);
+    checkbox.addEventListener('click', () => {
+      updateStatus(index, this.tasksList);
+      this.updateDisplay();
+    });
+    return item;
+  }
+
+  addTask(description) {
+    const taskObj = {
+      description,
+      completed: false,
+      index: this.tasksList.length,
+    };
+    this.tasksList.push(taskObj);
+    localStorage.setItem('tasks', JSON.stringify(this.tasksList));
+  }
+
+  editTask(newDesc, index) {
+    this.tasksList[index].description = newDesc;
+    localStorage.setItem('tasks', JSON.stringify(this.tasksList));
+  }
+
+  updateDisplay() {
+    const tasks = document.querySelector('ul');
+    tasks.innerHTML = '';
+    for (let i = 1; i < this.tasksList.length; i += 1) {
+      tasks.append(this.createTaskElement(this.tasksList[i]));
+    }
+    this.updateClearButton();
+  }
+
+  removeTask(index) {
+    this.tasksList.splice(index, 1);
+    for (let i = index; i < this.tasksList.length; i += 1) {
+      this.tasksList[i].index = i;
+    }
+    localStorage.setItem('tasks', JSON.stringify(this.tasksList));
+    this.updateDisplay();
+  }
```

Comment:
- [ ] Kindly make sure the removing functionality works as expected. At this moment your app is not performing the delete functionality correctly. Kindly fix that.

   - ### Your website behavior

   ![chrome-capture-2022-10-12 (6)](https://user-images.githubusercontent.com/87197412/201479866-ea6a9d2f-383a-4729-904a-61b7b7537aba.gif)